File: algorithm/baekjoon/codeplus/first-solve/src/main/java/com/keunwon/algorithm/greedy/Problem1789.java
package com.keunwon.algorithm.greedy;

import java.util.Scanner;

public class Problem1789 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        long value = sc.nextLong();

        long sum = 0L;
        int count = 0;
        while (sum <= value) {
            count++;
            sum += count;
        }

        System.out.println(count - 1);
    }
}

File: algorithm/kotlin-algorithm/src/main/kotlin/com/keunwon/algorithm/baekjoon/Problem1789.kt
package com.keunwon.algorithm.baekjoon

/**
 * Title: 수들의 합
 * Level: 실버-5
 **/
class Problem1789 {
    fun solution(n: Long): Int {
        var count = 0
        var sum = 0L

        while (sum <= n) {
            count++
            sum += count
        }
        return count - 1
    }
}

fun main() {
    val n = readLine()!!.toLong()
    Problem1789().solution(n).also(::println)
}
